api/preprints/serializers.py

```diff
@@ -530,3 +530,45 @@ def update(self, instance, validated_data):
     links = LinksField({
         'self': 'get_self_url',
     })
+
+
+class PreprintsInstitutionsRelationshipSerializer(BaseAPISerializer):
+    from api.institutions.serializers import InstitutionRelated  # Avoid circular import
+    data = ser.ListField(child=InstitutionRelated())
+
+    links = LinksField({
+        'self': 'get_self_url',
+        'html': 'get_related_url',
+    })
+
+    def get_self_url(self, obj):
+        return obj['self'].absolute_api_v2_url
+
+    def get_related_url(self, obj):
+        return f"{obj['self'].absolute_api_v2_url}institutions/"
+
+    class Meta:
+        type_ = 'institutions'
+
+    def make_instance_obj(self, obj):
+        return {
+            'data': obj.affiliated_institutions.all(),
+            'self': obj,
+        }
+
+    def update(self, instance, validated_data):
+        preprint = instance['self']
+        user = self.context['request'].user
+        update_institutions_if_user_associated(preprint, validated_data['data'], user)
```

Comment:
Related to my previous question, can `update_institutions()` do the job?

Reply:
It doesn't check the users affiliations so any user could remove all affiliations, not just the ones they have. I have to make some small changes.